nilearn/input_data/nifti_spheres_masker.py

```diff
@@ -85,6 +86,56 @@ def _apply_mask_and_get_affinity(seeds, niimg, radius, allow_overlap,
     return X, A
 
 
+def _apply_mask_get_rows(seeds, affine, target_shape, radius, allow_overlap,
+                                 mask_img=None):
+    seeds = list(seeds)
+
+    # Compute world coordinates of all voxels. The mask will be applied at 
+    # inversion time
+    mask_coords = list(np.ndindex(*target_shape))
+
+    nearests = []
+    for sx, sy, sz in seeds:
+        nearest = np.round(coord_transform(sx, sy, sz, np.linalg.inv(affine)))
+        nearest = nearest.astype(int)
+        nearest = (nearest[0], nearest[1], nearest[2])
+        try:
+            nearests.append(mask_coords.index(nearest))
+        except ValueError:
+            nearests.append(None)
+
+    mask_coords = np.asarray(list(zip(*mask_coords)))
+    mask_coords = coord_transform(mask_coords[0], mask_coords[1],
+                                  mask_coords[2], affine)
+    mask_coords = np.asarray(mask_coords).T
+
+    clf = neighbors.NearestNeighbors(radius=radius)
+    A = clf.fit(mask_coords).radius_neighbors_graph(seeds)
```

Comment:
Could we call it 'adjacency' rather than 'A' ?

Reply:
Yes, also for some of the points below: I was just repurposing the other code for the extraction of sphere's data. So I basically copied a lot over from there, which was involved in the creation of spheres.